### DISCUSSION TITLE
allgather doesn't concatenate tensors

### DISCUSSION THREAD

**Discussion by chandana1332** [horovod/horovod#2756] [Q&A]
**Environment:**
1. Framework: MXNet
2. Framework version: 1.4.1
3. Horovod version: 0.16.4
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
I'm running this to test allgather:

test = mx.ndarray.ones((2,3,2))
print(test)
new_test = hvd.allgather(test)
print(new_test)

If there are 4 processes, the expected shape of new_test is (8,3,2)

But the behaviour I notice is that the output shape is (2,3,2)

Is the usage of allgather accurate? If not, can you help figure out what might be wrong?

**Comment by apeforest** [comment]
Hi @chandana1332, allgather in Horovod with MXNet was not fully implemented and tested. There was a recent [PR](https://github.com/horovod/horovod/pull/1424) that provided a fix. Could you please try to see if it addressed your problem? Thanks

**Comment by chandana1332** [comment]
The PR was not merged so I can't build and test it.
I'm not too familiar with the internals of Horovod to verify if diff solves the problem I'm seeing. 

Do you have suggestions on how I can verify this fix?

**Comment by apeforest** [comment]
@chandana1332 I will provide you the steps offline
